lang/cs/Org.Apache.REEF.Driver/Task/DriverTaskState.cs

```diff
@@ -0,0 +1,129 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Org.Apache.REEF.Driver.Task
+{
+    /// <summary>
+    /// DriverTaskState wraps current state and provides methods to move from one state to the next state
+    /// </summary>
+    public sealed class DriverTaskState
+    {
+        internal static IDictionary<TaskStateTransition, TaskTransitionState> Transitions = new Dictionary<TaskStateTransition, TaskTransitionState>
+        {
+            { new TaskStateTransition(TaskTransitionState.TaskNew, TaskEvent.SubmittedTask), TaskTransitionState.TaskSubmitting },
+            { new TaskStateTransition(TaskTransitionState.TaskNew, TaskEvent.ClosedTask), TaskTransitionState.TaskClosedByDriver },            
+            { new TaskStateTransition(TaskTransitionState.TaskSubmitting, TaskEvent.RunningTask), TaskTransitionState.TaskRunning },
+            { new TaskStateTransition(TaskTransitionState.TaskSubmitting, TaskEvent.FailedTaskAppError), TaskTransitionState.TaskFailedAppError },
+            { new TaskStateTransition(TaskTransitionState.TaskSubmitting, TaskEvent.FailedTaskSystemError), TaskTransitionState.TaskFailedSystemError },
+            { new TaskStateTransition(TaskTransitionState.TaskSubmitting, TaskEvent.FailedTaskEvaluatorError), TaskTransitionState.TaskFailedByEvaluatorFailure },
+            { new TaskStateTransition(TaskTransitionState.TaskSubmitting, TaskEvent.FailedTaskCommunicationError), TaskTransitionState.TaskFailedByGroupCommunication },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.CompletedTask), TaskTransitionState.TaskCompleted },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.WaitingTaskToClose), TaskTransitionState.TaskWaitingForClose },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.FailedTaskAppError), TaskTransitionState.TaskFailedAppError },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.FailedTaskSystemError), TaskTransitionState.TaskFailedSystemError },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.FailedTaskEvaluatorError), TaskTransitionState.TaskFailedByEvaluatorFailure },
+            { new TaskStateTransition(TaskTransitionState.TaskRunning, TaskEvent.FailedTaskCommunicationError), TaskTransitionState.TaskFailedByGroupCommunication },
+            { new TaskStateTransition(TaskTransitionState.TaskWaitingForClose, TaskEvent.ClosedTask), TaskTransitionState.TaskClosedByDriver },
+            { new TaskStateTransition(TaskTransitionState.TaskWaitingForClose, TaskEvent.FailedTaskAppError), TaskTransitionState.TaskFailedAppError },
```

Comment:
I have a question/comment here: while the task gets closed when driver ordered it, it might be in some intermediate state and user might not think of logic to shut down task properly in these intermediate states...so u might get recoverable group comm. or system or app. error at this stage...So I am a little confused what to do here